Source/Integrations/KingfisherIntegration.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  KingfisherIntegration.swift
+//  AXPhotoViewer
+//
+//  Created by Matt Lamont on 11/08/17.
+//  Copyright © 2017 Matt Lamont. All rights reserved.
+//
+
+import Kingfisher
+
+class KingfisherIntegration: NSObject, NetworkIntegrationProtocol {
+
+    weak public var delegate: NetworkIntegrationDelegate?
+
+    fileprivate var downloadTasks = NSMapTable<PhotoProtocol, RetrieveImageTask>(keyOptions: .strongMemory, valueOptions: .strongMemory)
+
+    public func loadPhoto(_ photo: PhotoProtocol) {
+        if photo.imageData != nil || photo.image != nil {
+            delegate?.networkIntegration(self, loadDidFinishWith: photo)
+        }
+
+        guard let url = photo.url else {
+            return
+        }
+
+        let progress: DownloadProgressBlock = { [weak self] (receivedSize, totalSize) in
+            guard let uSelf = self else {
+                return
+            }
+
+            uSelf.delegate?.networkIntegration?(uSelf, didUpdateLoadingProgress: CGFloat(totalSize / receivedSize), for: photo)
```

Comment:
`CGFloat(totalSize / receivedSize)` will always evaluate to zero since `totalSize` and `receivedSize` are Integers

`CGFloat(totalSize) / CGFloat(receivedSize)` is the correct way of expressing this